bigquery/gcloud/aio/bigquery/bigquery.py

```diff
@@ -503,3 +527,64 @@ async def __aenter__(self) -> 'Table':
 
     async def __aexit__(self, *args: Any) -> None:
         await self.close()
+
+
+class Dataset(BigqueryBase):
```

Comment:
This file is starting to become quite the monolith with all these new classes. Would you mind splitting it apart? We should probably pull out the relevant classes to, at least, `job.py`, `table.py`, and `dataset.py` to help keep this understandable.

Reply:
Looks like this new class hasn't been exposed to users yet! Please check out the `__init__.py` file, where we define the public interface for users. You'll want to ensure this new `Dataset` class is included there!

Reply:
will do.

Reply:
Done.